journal/src/main/java/pl/wsztajerowski/journal/records/DoubleBatch.java

```diff
@@ -0,0 +1,152 @@
+package pl.wsztajerowski.journal.records;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class DoubleBatch {
+    private final ExecutorService executorService;
+    private final FileChannel writeChannel;
+    private final Batch batchA;
+    private final Batch batchB;
+    private final Condition batchIsFullCondition;
+    private volatile Batch currentBatch;
+    private final ReentrantLock batchLock;
+    private volatile boolean isClosed;
+
+
+    public static DoubleBatch open(Path journalFile, int batchSize) {
+        try {
+            ExecutorService executor = Executors.newSingleThreadExecutor(r -> new Thread(r, "write-channel-executor"));
+            FileChannel writerChannel = FileChannel.open(journalFile, StandardOpenOption.WRITE, StandardOpenOption.APPEND);
+
+            return new DoubleBatch(executor, writerChannel, batchSize);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    DoubleBatch(ExecutorService executor, FileChannel writeChannel, int batchSize) throws IOException {
```

Comment:
weird, you have static create method, and you initialize some of the objects, and others are initized in here, avoid work in constructors